apps/dashboard/src/components/side-navigation/usage-card.tsx

```diff
@@ -0,0 +1,120 @@
+import { useTelemetry } from '@/hooks/use-telemetry';
+import { ROUTES } from '@/utils/routes';
+import { TelemetryEvent } from '@/utils/telemetry';
+import { GetSubscriptionDto } from '@novu/shared';
+import { format } from 'date-fns';
+import { RiCalendarEventLine, RiErrorWarningLine } from 'react-icons/ri';
+import { Link } from 'react-router-dom';
+import { Button } from '../primitives/button';
+import { Progress } from '../primitives/progress';
+
+type UsageStatus = {
+  progressVariant: 'error' | 'warning' | 'default';
+  isComplete: boolean;
+};
+
+export type UsageCardProps = {
+  subscription?: GetSubscriptionDto;
+};
+
+export function UsageCard({ subscription }: UsageCardProps) {
+  const track = useTelemetry();
+  const currentEvents = subscription?.events?.current ?? 0;
+  const maxEvents = subscription?.events?.included ?? 10000;
+  const resetDate = subscription?.currentPeriodEnd ?? null;
+
+  const handleUsageCardClick = () => {
+    track(TelemetryEvent.USAGE_CARD_CLICKED, {
+      currentEvents,
+      maxEvents,
+      usagePercentage: getUsagePercentage(currentEvents, maxEvents),
+      isLimitReached: getUsageStatus(currentEvents, maxEvents).isComplete,
+    });
+  };
+
+  return (
+    <Link
+      to={ROUTES.SETTINGS_BILLING}
+      className="bg-bg-white group relative mb-2 flex h-[58px] cursor-pointer flex-col rounded-lg"
+      onClick={handleUsageCardClick}
+    >
+      <CardContent
+        currentEvents={currentEvents > maxEvents ? maxEvents : currentEvents}
+        maxEvents={maxEvents}
+        resetDate={resetDate}
+      />
+    </Link>
+  );
+}
+
+const formatNumber = (num: number): string =>
+  num >= 1000 ? `${(num / 1000).toFixed(1).replace(/\.0$/, '')}k` : num.toLocaleString();
+
+const getUsagePercentage = (current: number, limit: number): number => Math.min((current / limit) * 100, 100);
+
+const getUsageStatus = (current: number, limit: number): UsageStatus => {
+  const percentage = getUsagePercentage(current, limit);
+  const isComplete = percentage >= 100;
+
+  return {
+    progressVariant: percentage >= 80 ? 'error' : 'default',
+    isComplete,
+  };
+};
+
+type CardContentProps = {
+  currentEvents: number;
+  maxEvents: number;
+  resetDate: string | null;
+};
+
+function CardContent({ currentEvents, maxEvents, resetDate }: CardContentProps) {
+  const percentage = getUsagePercentage(currentEvents, maxEvents);
+  const { progressVariant, isComplete } = getUsageStatus(currentEvents, maxEvents);
+  const formattedResetDate = resetDate ? format(new Date(resetDate), 'MMM d yyyy') : '';
+
+  return (
+    <div className="relative flex flex-col overflow-hidden p-2">
```

Comment:
🎥 How about adding a subtle pop-in animation?

Reply:
I use subtle hover animation with tailwind there already 🚀

Reply:
Weird I didn't see it in the preview. Anyway, all good.